libraries/Microsoft.Bot.Builder.AI.LUIS/LuisRecognizerOptions.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Builder.AI.Luis
+{
+    public abstract class LuisRecognizerOptions
+    {
+        protected LuisRecognizerOptions(LuisApplication application)
+        {
+            Application = application ?? throw new ArgumentNullException(nameof(application));
+        }
+
+        public LuisApplication Application { get; set; }
+
+        /// <summary>
+        /// Gets or sets the time in milliseconds to wait before the request times out.
+        /// </summary>
+        /// <value>
+        /// The time in milliseconds to wait before the request times out. Default is 100000 milliseconds.
+        /// </value>
+        /// <remarks>
+        /// This value can only be set when <see cref="LuisRecognizer"/> is created and can't be changed
+        /// in individual <see cref="IRecognizer.RecognizeAsync"/> calls.
+        /// </remarks>
+        public double Timeout { get; set; } = 100000;
```

Comment:
Applies to this and all following properties:
Given the comments, consider making the setter private.
Might need a way to set those from the constructors at that point however.